apps/wallet/src/shared/utils/from-exported-keypair.ts

```diff
@@ -2,17 +2,24 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { type ExportedKeypair, type Keypair } from '@mysten/sui.js/cryptography';
+import {
+	decodeSuiPrivateKey,
+	LEGACY_PRIVATE_KEY_SIZE,
+	PRIVATE_KEY_SIZE,
+} from '@mysten/sui.js/cryptography/keypair';
 import { Ed25519Keypair } from '@mysten/sui.js/keypairs/ed25519';
 import { Secp256k1Keypair } from '@mysten/sui.js/keypairs/secp256k1';
 import { Secp256r1Keypair } from '@mysten/sui.js/keypairs/secp256r1';
-import { fromB64 } from '@mysten/sui.js/utils';
 
-const PRIVATE_KEY_SIZE = 32;
-const LEGACY_PRIVATE_KEY_SIZE = 64;
+export function validateExportedKeypair(keypair: ExportedKeypair): ExportedKeypair {
+	const _kp = decodeSuiPrivateKey(keypair.privateKey);
+	return keypair;
+}
+
 export function fromExportedKeypair(keypair: ExportedKeypair): Keypair {
-	const secretKey = fromB64(keypair.privateKey);
+	const { schema, secretKey } = decodeSuiPrivateKey(keypair.privateKey);
```

Comment:
@joyqvq the keypair here can be either in the new encoding or the previous one (base64). For example the test is failing because the existing imported keys in the storage are in base64 and this `decodeSuiPrivateKey` fails.

Adding a backwards compatible mode to the function so it tries to detect the encoding first, if possible, I thing it will fix the issue.

The issue is only with imported accounts because we store the private key encrypted in storage. If we add this backwards compatible mode [here](https://github.com/MystenLabs/sui/blob/privkey/apps/wallet/src/background/accounts/ImportedAccount.ts#L130) and [here](https://github.com/MystenLabs/sui/blob/privkey/apps/wallet/src/background/legacy-accounts/LegacyVault.ts#L56) I think it will fix it.

Reply:
If I get it right with your changes the ExportedKeyPair now has the schema and private key in the `privateKey` field of the exported keyPair. Maybe it would make sense to just make `ExportedKeyPair` be a string and handle as a special case the stored imported keypair that can be also in the previous format.